DDF-1610: Cometd Sample

oconnormi opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/321)
<!-- Reviewable:end -->

oconnormi commented:
@lessarderic
@figliold
@garrettfreibott
@andrewkfiedler
<!-- Reviewable comment -K2grBKdWPdPNRR4TBfD:1420240831 -->

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 10 of 11 files at r1.
Review status: 10 of 11 files reviewed at latest revision, 8 unresolved discussions.

---

<sup>**[distribution/sdk/sample-cometd/bower.json, line 5 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hC8NO0YYsDs-v0Crq)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/bower.json#L5)):</sup>
Do we need the author property here? We usually avoid people's names and emails in the source code.
<!-- Reviewable comment -K2hC8NO0YYsDs-v0Crr:-139860013 -->

---

<sup>**[distribution/sdk/sample-cometd/Gruntfile.js, line 30 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hE5F-46e2-dmXphUF)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/Gruntfile.js#L30)):</sup>
Should we add `**` in those paths in case someone adds sub-directories in the future?
<!-- Reviewable comment -K2hDu452c1qJ6aiv9xj:-193882154 -->

---

<sup>**[distribution/sdk/sample-cometd/package.json, line 17 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hER4aOTu9t03ZoW51)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/package.json#L17)):</sup>
Do we really need the `^` in the versions? Can't we just lock them in?
<!-- Reviewable comment -K2hER4aOTu9t03ZoW52:-354796764 -->

---

<sup>**[distribution/sdk/sample-cometd/README.md, line 7 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hFettKr3qqN3a8TEj)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/README.md#L7)):</sup>
Missing `cp` in the command.
<!-- Reviewable comment -K2hFettKr3qqN3a8TEk:-1012431521 -->

---

<sup>**[distribution/sdk/sample-cometd/src/main/webapp/index.html, line 22 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hFwXB2Fp-vakJFf_J)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/index.html#L22)):</sup>
Interesting. Had never seen that `section` element before.
<!-- Reviewable comment -K2hFwXB2Fp-vakJFf_K:-717004092 -->

---

<sup>**[distribution/sdk/sample-cometd/src/main/webapp/index.html, line 33 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hGG2HJbkJMF6WZQD2)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/index.html#L33)):</sup>
Probably can delete that commented out line ;-)
<!-- Reviewable comment -K2hGG2HJbkJMF6WZQD3:-1598688309 -->

---

<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 59 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hGr4k4EhdMmq93oj3)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L59)):</sup>
Do you know if the `cometd.batch()` call is needed? I think I was able to get things working without it. If we keep it, maybe we should put a comment explaining why it's needed.
<!-- Reviewable comment -K2hGr4k4EhdMmq93oj4:1068142932 -->

---

<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hHHFuFuNH_9PA-GRX)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L70)):</sup>
Is the `publish` needed to flush out the messages already in the queue? If so, should we document that?
<!-- Reviewable comment -K2hHHFuFuNH_9PA-GRY:-272477075 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

stustison commented on distribution/sdk/sample-cometd/bower.json:
```diff
@@ -0,0 +1,22 @@
+  "description": "Example of how to interact with ddf over cometd",
+  "main": "index.js",
+  "authors": [
+    "Michael O'Connor <michael.oconnor@connexta.com>"
```
Leave off the authors tag here. We've been slowly removing them all as we go along.

stustison commented on distribution/sdk/sample-cometd/package.json:
```diff
@@ -0,0 +1,24 @@
+{
```
Might want to set "private": true

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hHHFuFuNH_9PA-GRX)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L70)):</sup>
Yep, publishing an empty message causes the server to send any persisted messages to the client
<!-- Reviewable comment -K2hIoiXkn4Rg6Wn55zH:-2078734468 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hHHFuFuNH_9PA-GRX)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L70)):</sup>
@lessarderic, now I just need to find the right place to document it, although I want to say that was in the documentation somewhere as I'm thinking that's how I knew about it in the first place
<!-- Reviewable comment -K2hJ1cCVz2phAwhKVZm:639416573 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/index.html, line 22 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hFwXB2Fp-vakJFf_J)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/index.html#L22)):</sup>
It's an html5 element, there are several new tags coming in that will replace the endless numbers of divs used by most pages in favor of a more semantic page structure
<!-- Reviewable comment -K2hK7wvCCwMPrSlPGj7:252604092 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 1 of 11 files at r1.
Review status: 10 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 187 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hK29_sZSIBAgenwmy)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L187)):</sup>
Should we define the base URL at the top of the file and use it to build the other URLs? Should we also put a note somewhere (README.md maybe?) saying that the sample assumes DDF is using the default hostname and port?
<!-- Reviewable comment -K2hK29_sZSIBAgenwmz:1170023425 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Are you going to update the existing asciidocs in a separate PR?
<!-- Reviewable comment -K2hKnVwKr84j_RyIf5X:1736063722 -->

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hHHFuFuNH_9PA-GRX)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L70)):</sup>
When I said "document", I only meant update the comment :-)
<!-- Reviewable comment -K2hL718oDLetikfiN-q:-1979934555 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/package.json, line 17 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hER4aOTu9t03ZoW51)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/package.json#L17)):</sup>
We could, though as dev dependencies its less important, also that was just the default way that npm adds them when using either `npm install --save foo` or `npm install --save-dev foo`
<!-- Reviewable comment -K2hLB48Sq_Pf9L-LmQf:-1199509378 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/package.json, line 1 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2gUtT1XoOCf-4DQ8LN-r1-1)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/package.json#L1)):</sup>
Good call
<!-- Reviewable comment -K2hLhG279Ma1nGDTSH0:-1849837476 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 187 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hK29_sZSIBAgenwmy)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L187)):</sup>
Alternatively, I could also make it a text box on the page and have it default to the current url. Thoughts?
<!-- Reviewable comment -K2hNHGGTw-CR2Rytmyi:-1891100883 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 59 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hGr4k4EhdMmq93oj3)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L59)):</sup>
Added a comment
<!-- Reviewable comment -K2hQ7xshG6TeIMhofDt:-1276660037 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/package.json, line 1 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2gUtT1XoOCf-4DQ8LN-r1-1)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/package.json#L1)):</sup>
Done.
<!-- Reviewable comment -K2hR6r0yA1-V6zag405:-499146009 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 6 of 6 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[distribution/sdk/sample-cometd/package.json, line 17 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hER4aOTu9t03ZoW51)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/package.json#L17)):</sup>
:+1: 
<!-- Reviewable comment -K2hp3RYD4Fny_k7kAeK:-510466975 -->

---

<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 187 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hK29_sZSIBAgenwmy)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L187)):</sup>
That'd be nicer for sure, but I'm not sure how much more time we want to spend improving this. Up to you. I'm fine either way.
<!-- Reviewable comment -K2hp9LvvowH4WD50I3-:269234336 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
I will add them to this PR, wasn't sure exactly where to put them though, I think it would be part of the Extending or Integrating doc under the ddf/catalog/ui/docs section. There doesn't seem to be much on either of those pages currently
<!-- Reviewable comment -K2lcrVkCN2R15TxC7Sy:996381353 -->

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[distribution/sdk/sample-cometd/src/main/webapp/js/application.js, line 187 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2hK29_sZSIBAgenwmy)** ([raw file](https://github.com/codice/ddf/blob/e5a196ff578f7e51a45867712e4e21d3f55f30a3/distribution/sdk/sample-cometd/src/main/webapp/js/application.js#L187)):</sup>
I just added a comment to the readme regarding using other urls, and moved the `var cometURL` up to the top of the `application.js` file
<!-- Reviewable comment -K2lk-nT98fTkMuWNVi8:-44255647 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
:+1: 
<!-- Reviewable comment -K2reioDtDD1yd0_jLbe:-510466975 -->

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[catalog/ui/docs/src/main/resources/IntContents.adoc, line 9 \[r5\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2reYbhYnxyjpQVhwbp)** ([raw file](https://github.com/codice/ddf/blob/a079f004d145427ae00149866e5b33e56144d1e8/catalog/ui/docs/src/main/resources/IntContents.adoc#L9)):</sup>
I'd refer to the SDK for an example. Other than that lgtm.
<!-- Reviewable comment -K2reYbhYnxyjpQVhwbq:-1824532651 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
<sup>**[catalog/ui/docs/src/main/resources/IntContents.adoc, line 9 \[r5\]](https://reviewable.io:443/reviews/codice/ddf/321#-K2reYbhYnxyjpQVhwbp)** ([raw file](https://github.com/codice/ddf/blob/a079f004d145427ae00149866e5b33e56144d1e8/catalog/ui/docs/src/main/resources/IntContents.adoc#L9)):</sup>
Just added it
<!-- Reviewable comment -K2rhGy45cplOE4a1QT9:-467409926 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

oconnormi commented:
Reviewed 4 of 11 files at r1, 5 of 6 files at r2.
Review status: 11 of 12 files reviewed at latest revision, 2 unresolved discussions.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/321)
<!-- Sent from Reviewable.io -->
